fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/client/item/v1/CustomModeledArmor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.item.v1;
+
+import net.minecraft.client.render.entity.model.BipedEntityModel;
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Implement this interface on an {@link net.minecraft.item.ArmorItem} which you would like to have a custom armor model.
```

Comment:
I dont think we should really have what should be client only interfaces like this, Vanilla is moving away from this (think render layers) a registry would be better IMO.

Reply:
think it's easier to have an interface, for example: `BlockEntityClientSerializable`, you don't want to have an registry for such a thing. since the armor models and textures might depend on the item itself, I think it is more simple and easier to have it be an interface.

Reply:
BlockEntityClientSerializable is for the server tho. A vanilla example of this is the BlockEntityRenderers, there is a clear speration between client and server code, without relaying on the annotations. A registry might make it easier for a mod to change the model of lets say the vanilla armor without requiring a mixin.

Reply:
hmm true, I agree that it will be easier to change the vanilla armor that way. I will change this into a registry